pkg/executor/insert.go

```diff
@@ -509,19 +511,22 @@ func (e *InsertExec) initEvalBuffer4Dup() {
 	// Use public columns for new row.
 	numCols := len(e.Table.Cols())
 	// Use writable columns for old row for update.
-	numWritableCols := len(e.Table.WritableCols())
+	numWritableCols := len(e.Table.WritableCols()) + 1
 
 	extraLen := 0
 	if e.SelectExec != nil {
 		extraLen = e.SelectExec.Schema().Len() - e.rowLen
 	}
 
-	evalBufferTypes := make([]*types.FieldType, 0, numCols+numWritableCols+extraLen)
+	evalBufferTypes := make([]*types.FieldType, 0, numCols+numWritableCols+extraLen+1)
 
 	// Append the old row before the new row, to be consistent with "Schema4OnDuplicate" in the "Insert" PhysicalPlan.
 	for _, col := range e.Table.WritableCols() {
 		evalBufferTypes = append(evalBufferTypes, &(col.FieldType))
 	}
+	commitTSFt := types.NewFieldType(mysql.TypeLonglong)
+	commitTSFt.SetFlag(mysql.UnsignedFlag)
+	evalBufferTypes = append(evalBufferTypes, commitTSFt)
```

Comment:
For the same reason, I think not all statements need commit ts to reduce performance cost